cookbooks/cdo-varnish/libraries/helpers.rb

```diff
@@ -0,0 +1,96 @@
+# Various Ruby helper methods to generate Varnish VCL from the base configuration.
+
+# Basic regex matcher for the query part of a URL.
+QUERY_REGEX = "(\\?[a-z0-9]+)?$"
+
+def path_to_regex(path)
+  path = "^/#{path.sub(/^\//,'')}"
```

Comment:
This might be easier to read as:

path = "/#{path}" unless path.start_with?('/')